components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/BuildCmd.java

```diff
@@ -93,10 +93,11 @@ public void execute() {
         //extract the ballerina platform and runtime
         ToolkitLibExtractionUtils.extractPlatformAndRuntime();
 
-        File importedAPIDefLocation = new File(GatewayCmdUtils.getProjectAPIDefinitionsDirectoryPath(projectName));
+        File importedAPIDefLocation = new File(GatewayCmdUtils.getProjectGenAPIDefinitionPath(projectName));
         File addedAPIDefLocation = new File(GatewayCmdUtils.getProjectAPIFilesDirectoryPath(projectName));
 
-        if (importedAPIDefLocation.list() != null && importedAPIDefLocation.list().length == 0
+        if (((importedAPIDefLocation.list() != null && importedAPIDefLocation.list().length == 0) ||
+                importedAPIDefLocation.list() == null)
```

Comment:
Same logic should apply to `addedAPIDefLocation.list()` aswell
Also the logic in this if block looks too complex for one condition. Shall we break it down to two simple if blocks using `isValidImportedAPIDefLocation` and `isValidAddedAPIDefLocation` boolean variables